src/Fixer/PhpUnit/PhpUnitSetUpTearDownVisibilityFixer.php

```diff
@@ -0,0 +1,165 @@
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Gert de Pagter
+ */
+final class PhpUnitSetUpTearDownVisibilityFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Changes the visibility of the setUp and tearDown functions of phpunit to protected, to match its parent.',
+            [
+                new CodeSample(
+                    '<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+    private $hello;
+    public function setUp()
+    {
+        $this->hello = "hello";
+    }
+
+    public function tearDown()
+    {
+        $this->hello = null;
+    }
+}
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_FUNCTION]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach (array_reverse($this->findPhpUnitClasses($tokens)) as $indexes) {
+            $this->fixSetUpAndTearDown($tokens, $indexes[0], $indexes[1]);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     *
+     * @return int[][] array of [start, end] indexes from sooner to later classes
+     */
+    private function findPhpUnitClasses(Tokens $tokens)
+    {
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
+        $phpunitClasses = [];
+
+        for ($index = 0, $limit = $tokens->count() - 1; $index < $limit; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_CLASS) && $phpUnitTestCaseIndicator->isPhpUnitClass($tokens, $index)) {
+                $index = $tokens->getNextTokenOfKind($index, ['{']);
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $phpunitClasses[] = [$index, $endIndex];
+                $index = $endIndex;
+            }
+        }
+
+        return $phpunitClasses;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function fixSetUpAndTearDown(Tokens $tokens, $startIndex, $endIndex)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        for ($i = $endIndex - 1; $i > $startIndex; --$i) {
+            if (!$this->isMethod($tokens, $i)) {
+                continue;
+            }
+            if (!$this->isMethodSetUpOrTearDown($tokens, $i)) {
+                continue;
+            }
+
+            $visibility = $tokensAnalyzer->getMethodAttributes($i)['visibility'];
+
+            if (T_PROTECTED === $visibility) {
+                continue;
+            }
+
+            if (T_PUBLIC === $visibility) {
+                do {
+                    $i = $tokens->getPrevMeaningfulToken($i);
+                } while (!$tokens[$i]->isGivenKind(T_PUBLIC));
```

Comment:
you can use `$tokens->getPrevTokenOfKind` to find the `T_PUBLIC` as replacement of this loop